ratis-common/src/main/java/org/apache/ratis/datastream/SupportedDataStreamType.java

```diff
@@ -20,30 +20,40 @@
 import org.apache.ratis.conf.Parameters;
 import org.apache.ratis.util.ReflectionUtils;
 
-public enum SupportedDataStreamType implements DataStreamFactory {
-  DISABLED("org.apache.ratis.server.impl.DisabledDataStreamFactory"),
+public enum SupportedDataStreamType implements DataStreamType {
+  DISABLED("org.apache.ratis.client.DisabledDataStreamClientFactory",
+      "org.apache.ratis.server.DisabledDataStreamServerFactory"),
   NETTY("org.apache.ratis.netty.NettyDataStreamFactory");
```

Comment:
@szetszwo Do we also need to split NettyDataStreamFactory into client and server ?

Reply:
Netty is in ratis-netty which depends on ratis-server and ratis-client.  So splitting it won't be useful at the moment.  It will be useful if we split ratis-netty into ratis-netty-client and ratis-netty-server later on.